fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/SearchServiceImpl.java

```diff
@@ -29,9 +42,69 @@
  */
 @Component
 public class SearchServiceImpl implements SearchService {
+    private static final String FEDORA_ID_DB_COLUMN = "fedora_id";
+
+    @Inject
+    private DataSource dataSource;
+
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Inject
+    private PlatformTransactionManager platformTransactionManager;
+
+    /**
+     * Connect to the database
+     */
+    @PostConstruct
+    private void setup() {
+        jdbcTemplate = getNamedParameterJdbcTemplate();
+    }
+
+    private NamedParameterJdbcTemplate getNamedParameterJdbcTemplate() {
+        return new NamedParameterJdbcTemplate(this.dataSource);
+    }
 
     @Override
-    public SearchResult doSearch(final SearchParameters parameters) {
-        return new SearchResult();
+    public SearchResult doSearch(final SearchParameters parameters) throws InvalidQueryException {
+        //translate parameters into a SQL query
+        final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+        final var whereClauses = new ArrayList<String>();
+        for (Condition condition : parameters.getConditions()) {
+            if (condition.getField().equals(Condition.Field.fedora_id) &&
+                    condition.getOperator().equals(Condition.Operator.EQ)) {
+                var object = condition.getObject();
+                if (!object.equals("*")) {
+                    if (object.contains("*")) {
+                        object = object.replace("*", "%");
+                        whereClauses.add(FEDORA_ID_DB_COLUMN + " like '" + object + "'");
+                    } else {
+                        whereClauses.add(FEDORA_ID_DB_COLUMN + " = '" + object + "'");
+                    }
+                }
+            } else {
+                throw new InvalidQueryException("Condition not supported: " +
```

Comment:
Maybe `Condition.toString()` should produce a string in this format?

Reply:
+1